### PR TITLE
MOI tests: intlinear

### PR DESCRIPTION
**Pull request by laradicp**
#436 `intlinear` tests

### CODE REVIEW

File: src/ColunaBase/solsandbounds.jl [hunk 1]
```diff
@@ -133,23 +133,23 @@ MOI [`TerminationStatusCode`](https://jump.dev/MathOptInterface.jl/stable/apiref
...
+- `OPTIMAL` : the algorithm found a global optimal solution given the optimality tolerance
 - `INFEASIBLE` : the algorithm proved infeasibility
 - `TIME_LIMIT` : the algorithm stopped because of the time limit
 - `NODE_LIMIT` : the branch-and-bound based algorithm stopped due to the node limit
 - `OTHER_LIMIT` : the algorithm stopped because of a limit that is neither the time limit 
 nor the node limit
 
 If the algorithm has not been called, the default value of the termination status should be:
-- `UNKNOWN_TERMINATION_STATUS`
+- `OPTIMIZE_NOT_CALLED`
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:144]: Why did you rename this constant ?

Comment (laradicp) [reply src/ColunaBase/solsandbounds.jl:144]: I tried to use `UNKNOWN_TERMINATION_STATUS` as `MOI.OPTIMIZE_NOT_CALLED`, which we need for `intlinear` tests. But apparently `UNKNOWN_TERMINATION_STATUS` has another meaning and can't be converted to `MOI.OPTIMIZE_NOT_CALLED` (I got an error in another test that had nothing to do with MOI tests). So I created this new constant to use as a direct convertion of `MOI.OPTIMIZE_NOT_CALLED` to Coluna, and it fit better in the definition above than `UNKNOWN_TERMINATION_STATUS` after my changes. But I can come back to the error I got when using `UNKNOWN_TERMINATION_STATUS` as the Coluna reference to `MOI.OPTIMIZE_NOT_CALLED` and try to fix it, so we could drop `OPTIMIZE_NOT_CALLED` in Coluna.

Comment (guimarqu) [reply src/ColunaBase/solsandbounds.jl:144]: so if you convert `MOI.OPTIMIZE_NOT_CALLED` into `Coluna.UNKNOWN_TERMINATION_STATUS`, you have another error ? What's the error ?

Comment (laradicp) [reply src/ColunaBase/solsandbounds.jl:144]: Yes. I didn't look into it though, just assumed `UNKNOWN_TERMINATION_STATUS` had another purpose. But I can try to fix it.
```
[ Info: Terminated
[ Info: Primal bound: 1661.0
[ Info: Dual bound: 1547.3888889188888
User algorithms: Error During Test at /home/lara/Documentos/LAB/atoptima/Coluna.jl/test/user_algorithms_tests.jl:92
  Test threw exception
  Expression: JuMP.objective_value(model) < 2000.0
  OptimizeNotCalled()
  Stacktrace:
   [1] _moi_get_result(model::Coluna.Optimizer, args::MathOptInterface.ObjectiveValue)
     @ JuMP ~/.julia/packages/JuMP/Xrr7O/src/JuMP.jl:1077
   [2] get(model::Model, attr::MathOptInterface.ObjectiveValue)
     @ JuMP ~/.julia/packages/JuMP/Xrr7O/src/JuMP.jl:1100
   [3] objective_value(model::Model; result::Int64)
     @ JuMP ~/.julia/packages/JuMP/Xrr7O/src/objective.jl:42
   [4] objective_value
     @ ~/.julia/packages/JuMP/Xrr7O/src/objective.jl:42 [inlined]
   [5] conseq_colgen_test()
     @ Main ~/Documentos/LAB/atoptima/Coluna.jl/test/user_algorithms_tests.jl:92
   [6] user_algorithms_tests()
     @ Main ~/Documentos/LAB/atoptima/Coluna.jl/test/user_algorithms_tests.jl:6
   [7] macro expansion
     @ ~/Documentos/LAB/atoptima/Coluna.jl/test/runtests.jl:44 [inlined]
   [8] macro expansion
     @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [9] top-level scope
     @ ~/Documentos/LAB/atoptima/Coluna.jl/test/runtests.jl:44
Test Summary:   | Error  Total
User algorithms |     1      1
```

Comment (guimarqu) [reply src/ColunaBase/solsandbounds.jl:144]: `UNKNOWN_TERMINATION_STATUS` was supposed to be the equivalent of `OPTIMIZE_NOT_CALLED` but seems like this name was misleading so I agree to keep your change.

File: src/ColunaBase/solsandbounds.jl [hunk 2]
```diff
@@ -162,7 +162,7 @@ Description of the solution statuses:
 
 If there is no solution or if we don't have information about the solution, the 
 solution status should be :
-- `UNKWNOW_SOLUTION_STATUS`
+- `UNKNOWN_SOLUTION_STATUS`
```
Comment (guimarqu) [inline src/ColunaBase/solsandbounds.jl:165]: nice catch

File: src/MOIwrapper.jl [hunk 3]
```diff
@@ -587,6 +591,7 @@ function MOI.empty!(model::Coluna.Optimizer)
     if model.default_optimizer_builder !== nothing
         set_default_optimizer_builder!(model.inner, model.default_optimizer_builder)
     end
+    setterminationstatus!(model.result, OPTIMIZE_NOT_CALLED)
```
Comment (guimarqu) [inline src/MOIwrapper.jl:594]: Maybe you should reinitialize the whole result rather than just the termination status ?

File: src/MathProg/MOIinterface.jl [hunk 4]
```diff
@@ -100,7 +100,9 @@ function enforce_kind_in_optimizer!(form::Formulation, v::Variable)
     moirecord = getmoirecord(v)
     moi_kind = getkind(moirecord)
     if moi_kind.value != -1
-        MOI.delete(inner, moi_kind)
+        if MOI.is_valid(inner, moi_kind)
```
Comment (guimarqu) [inline src/MathProg/MOIinterface.jl:103]: what was the problem here ?

**Review by guimarqu — APPROVED** [review]

File: src/ColunaBase/solsandbounds.jl (hunk 1 above)
Comment (laradicp) [reply src/ColunaBase/solsandbounds.jl:144]: But should we keep `UNKNOWN_TERMINATION_STATUS` then? I mean, the error must be fixed if we replace it completely to `OPTIMIZE_NOT_CALLED`. It's still used in this function, I didn't try to change to `OPTIMIZE_NOT_CALLED` yet:
```julia
function run!(algo::SolveIpForm, env::Env, data::ModelData, input::OptimizationInput)::OptimizationOutput
    [...]
    if !ip_supported
        @warn "Optimizer of formulation with id =", getuid(form),
              " does not support integer variables. Skip SolveIpForm algorithm."
        setterminationstatus!(result, UNKNOWN_TERMINATION_STATUS)
        return OptimizationOutput(result)
    end
    [...]
```
Also it's still the default value here in `OptimizationState`, I got the same error I put in the comment above when I tried to change it to `OPTIMIZE_NOT_CALLED`.
```julia
function OptimizationState(
    form::F;
    termination_status::TerminationStatus = UNKNOWN_TERMINATION_STATUS,
    [...]
```

File: src/MOIwrapper.jl (hunk 3 above)
Comment (laradicp) [reply src/MOIwrapper.jl:594]: True

File: src/MathProg/MOIinterface.jl (hunk 4 above)
Comment (laradicp) [reply src/MathProg/MOIinterface.jl:103]: It tried to delete a constraint that didn't exist anymore. An error was thrown and I remember it was in the second optimization the MOI test made to the same model.

File: src/Algorithm/basic/solveipform.jl [hunk 5]
```diff
@@ -61,7 +61,6 @@ function run!(algo::SolveIpForm, env::Env, data::ModelData, input::OptimizationI
     if !ip_supported
         @warn "Optimizer of formulation with id =", getuid(form),
               " does not support integer variables. Skip SolveIpForm algorithm."
-        setterminationstatus!(result, UNKNOWN_TERMINATION_STATUS)
```
Comment (guimarqu) [inline src/Algorithm/basic/solveipform.jl:64]: why did you remove this line ?

Comment (laradicp) [reply src/Algorithm/basic/solveipform.jl:64]: Isn't it already set some lines above when `result` is declared? 
```julia
result = OptimizationState(
  form, 
  ip_primal_bound = get_ip_primal_bound(getoptstate(input)),
  max_length_ip_primal_sols = algo.max_nb_ip_primal_sols
)
```
`OPTIMIZE_NOT_CALLED` is the default value for `termination_status` in `OptimizationState`.

Comment (guimarqu) [reply src/Algorithm/basic/solveipform.jl:64]: Yes it is.
